### PR TITLE
Add gemspecs

### PR DESCRIPTION
**Pull request by magneland**

### CODE REVIEW

File: lib/organization_gem_dependencies.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # frozen_string_literal: true
 
+require 'organization_gem_dependencies/version_ranges_intersection'
```
Comment (bboe) [inline lib/organization_gem_dependencies.rb]: Sort?

File: lib/organization_gem_dependencies/cli.rb [hunk 2]
```diff
@@ -10,11 +10,15 @@
...
   # Define the command line interface.
   class Cli
+    using VersionRangesIntersection
+
     SEARCH_TERM = 'org:%s filename:Gemfile.lock'
     USAGE = <<~USAGE
       Usage: organization_gem_dependencies [options] GITHUB_ORGANIZATION
     USAGE
 
+    GEMSPEC_SEARCH_TERM = 'org:%s extension:gemspec'
```
Comment (bboe) [inline lib/organization_gem_dependencies/cli.rb]: Sort?

File: lib/organization_gem_dependencies/cli.rb [hunk 3]
```diff
@@ -50,6 +53,29 @@ def run
...
       end
+
+      gemfiles(github, github_organization, GEMSPEC_SEARCH_TERM) do |gemspec|
+        gemspec_path = "#{gemspec.repository.name}/#{gemspec.path}"
+        if filtered?(gemspec_path)
+          STDERR.puts "Skipping #{gemspec_path}"
+          next
+        end
+        STDERR.puts "Processing #{gemspec_path}"
+        content = nil
```
Comment (bboe) [inline lib/organization_gem_dependencies/cli.rb]: You removed this in the other block, should it be here too?

**Comment by magneland** [comment]
I am using a refinement or two here. I don't know if this belongs in Bundler itself.
I also struggled a bit with the naming, but I am trying to think mathematical here while conforming with general rubyist language.

File: lib/organization_gem_dependencies/cli.rb [hunk 4]
```diff
@@ -50,6 +53,29 @@ def run
           "#{gemfile.repository.name}/#{gemfile.path}"
         ))
       end
+
+      gemfiles(github, github_organization, GEMSPEC_SEARCH_TERM) do |gemspec|
```
Comment (bboe) [inline lib/organization_gem_dependencies/cli.rb]: This block is nearly identical to the above save for the call to `process_` and the search term. Can we dry this up?

File: lib/organization_gem_dependencies/cli.rb [hunk 5]
```diff
@@ -99,9 +125,9 @@ def filtered?(gemfile_path)
       false
     end
 
-    def gemfiles(github, organization)
+    def gemfiles(github, organization, search_term = SEARCH_TERM)
```
Comment (bboe) [inline lib/organization_gem_dependencies/cli.rb]: Can we explicitly pass the search term in both cases?

File: lib/organization_gem_dependencies/cli.rb [hunk 6]
```diff
@@ -160,16 +186,20 @@ def parse_options
       @options = { direct: false, ignore_paths: [] }
       OptionParser.new do |config|
         config.banner = USAGE
+        config.on('-h', '--help', 'Prints this help') do
```
Comment (bboe) [inline lib/organization_gem_dependencies/cli.rb]: Does option parser not have a built-in for help?

File: lib/organization_gem_dependencies/version_ranges_intersection.rb [hunk 7]
```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+module VersionRangesIntersection
+  refine Bundler::VersionRanges::ReqR do
+    # Checks whether two ranges overlap
+    def intersect?(other)
+      a = self
```
Comment (bboe) [inline lib/organization_gem_dependencies/version_ranges_intersection.rb]: Can we use better names?

File: lib/organization_gem_dependencies/cli.rb [hunk 8]
```diff
@@ -10,11 +10,15 @@
 module OrganizationGemDependencies
   # Define the command line interface.
   class Cli
+    using VersionRangesIntersection
+
     SEARCH_TERM = 'org:%s filename:Gemfile.lock'
```
Comment (bboe) [inline lib/organization_gem_dependencies/cli.rb]: Modify to GEMFILE_SEARCH_TERM?

File: lib/organization_gem_dependencies/cli.rb [hunk 9]
```diff
@@ -3,18 +3,22 @@
 require 'base64'
 require 'io/console'
 require 'json'
-require 'optparse'
-
 require 'octokit'
+require 'optparse'
```
Comment (bboe) [inline lib/organization_gem_dependencies/cli.rb]: While I don't think ruby style guide has any such recommendations, I follow the python import recommendations which are system packages first, then third party, then local separated by a newline.

With that, optparse should remain where it was, with oktokit and the last require line separated by a single newline.

File: lib/organization_gem_dependencies/cli.rb [hunk 10]
```diff
@@ -28,28 +32,17 @@ def run
...
-          STDERR.puts "Sleeping #{sleep_time} seconds"
-          sleep(sleep_time)
-          retry
-        end
-        merge!(gems, process_gemfile(
-          Bundler::LockfileParser.new(content),
-          "#{gemfile.repository.name}/#{gemfile.path}"
-        ))
+
+      remote_search(github, github_organization, GEMFILE_LOCK_SEARCH_TERM) do |gemfile_lock|
```
Comment (bboe) [inline lib/organization_gem_dependencies/cli.rb:38]: This is a fantastic improvement.

**Review by bboe — APPROVED** [review]
Looks awesome. One minor comment about require order, but it's me being picky.

**Pull request merged**